src/org/labkey/test/components/experiment/NodeDetailItem.java

```diff
@@ -0,0 +1,113 @@
+package org.labkey.test.components.experiment;
+
+import org.labkey.test.Locator;
+import org.labkey.test.components.Component;
+import org.labkey.test.components.WebDriverComponent;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+
+public class NodeDetailItem extends WebDriverComponent<NodeDetailItem.ElementCache>
+{
+    final WebElement _el;
+    final WebDriver _driver;
+
+    public NodeDetailItem(WebElement element, WebDriver driver)
+    {
+        _el = element;
+        _driver = driver;
+    }
+
+    public String getName()
+    {
+        if (isNameLinked())
+            return elementCache().nameLink.findElement(this).getText();
+        else
+            return elementCache().nameSpan.findElement(this).getText();
+    }
+
+    private boolean isNameLinked()
+    {
+        return elementCache().nameSpan.existsIn(this);
+    }
+
+    public void clickOverViewLink()
+    {
+        getWrapper().mouseOver(getComponentElement());
+        getWrapper().waitFor(()-> elementCache().overviewLink.isEnabled(), 1000);
+        getWrapper().clickAndWait(elementCache().overviewLink);
+    }
+
+    public void getLineageGraphLink()
+    {
+        getWrapper().mouseOver(getComponentElement());
+        getWrapper().waitFor(()-> elementCache().lineageGraphLink.isEnabled(), 1000);
+        getWrapper().clickAndWait(elementCache().lineageGraphLink);
+    }
+
+    @Override
+    public WebElement getComponentElement()
+    {
+        return _el;
+    }
+
+    @Override
+    public WebDriver getDriver()
+    {
+        return _driver;
+    }
+
+    @Override
+    protected ElementCache newElementCache()
+    {
+        return new ElementCache();
+    }
+
+
+    protected class ElementCache extends Component<?>.ElementCache
+    {
+        final WebElement image = Locator.tagWithClass("img", "lineage-sm-icon")
+                .findWhenNeeded(this);
+
+        final WebElement overviewLink = Locator.linkWithSpan("Overview")
+                .findWhenNeeded(this).withTimeout(2000);
+        final WebElement lineageGraphLink = Locator.linkWithSpan("Lineage")
+                .findWhenNeeded(this).withTimeout(2000);
+        final Locator nameLink = Locator.tagWithClass("a", "pointer");
+        final Locator nameSpan = Locator.tag("span");
```

Comment:
You can combine Locators rather than checking for one and falling back to the other.
```suggestion
        final WebElement nameEl = Locator.union(
                Locator.tagWithClass("a", "pointer"), 
                Locator.tag("span"))
            .findWhenNeeded(this);
```

Reply:
I can't resolve Locator.union from anywhere; suggestions?
I considered using Locator.findAnyElement(description, context, Locator... locators) but didn't like the idea of not really knowing which it is
I ended up with what's there because it worked